packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1741,12 +1752,24 @@ class SemanticsConfiguration {
     _setFlag(SemanticsFlags.isSelected, value);
   }
 
+  /// Whether the owning [RenderObject] is currently enabled.
+  ///
+  /// A disabled object does not respond to user interactions. Only objects that
+  /// usually do respond to user interactions but currently don't (like a
+  /// disabled button) should be marked as disabled. Objects (like static text)
+  /// that never respond to user interactions should not be marked as disabled.
+  bool get isEnabled => _hasFlag(SemanticsFlags.hasEnabledState) ? _hasFlag(SemanticsFlags.isEnabled) : null;
+  set isEnabled(bool value) {
+    _setFlag(SemanticsFlags.hasEnabledState, true);
+    _setFlag(SemanticsFlags.isEnabled, value);
+  }
+
   /// If this node has Boolean state that can be controlled by the user, whether
   /// that state is on or off, corresponding to true and false, respectively.
   ///
   /// Do not set this to any value if the owning [RenderObject] doesn't have
   /// Booleans state that can be controlled by the user.
-  bool get isChecked => _hasFlag(SemanticsFlags.hasCheckedState) && _hasFlag(SemanticsFlags.isChecked);
+  bool get isChecked => _hasFlag(SemanticsFlags.hasCheckedState) ? _hasFlag(SemanticsFlags.isChecked) : null;
```

Comment:
It would be really preferable not to have a boolean getter that can return null. Almost no one expects returned booleans to be tri-valued. If there's no way around this, then we should document very clearly that the result is null when there's no semantics checked state and that the result should always be null-checked. 

Another alternative would be to provide a means of checking whether this getter makes sense given the current flags, and make it a StateError to call this when it doesn't.

Reply:
When we get NNBD in Dart, this will become `bool?` (or if we don't get NNBD, then we're doomed, do-o-o-o-o-o-o-o-o-o-omed!)

Reply:
I kinda like it that this (and isEnabled) is tri-valued because it fits the use case here.

The alternative would be to split this in two and expose hasCheckedState for isChecked as well as hasEnabledState for isEnabled into its own getter/setter pair. That would make the usage of the API a little more complicated: before setting isEnabled you'd actually have to set hasEnabledState first. Same for isChecked.